node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/MetricsReporter.java

```diff
@@ -100,13 +101,17 @@ private void updateAllocationMetrics(NodeList nodes) {
                     nonActiveNodes++;
                 }
             }
+            AtomicReference<ClusterSpec.Type> clusterType = new AtomicReference<>();
+            allocatedNodes.stream().findFirst().flatMap(Node::allocation).ifPresent(allocation -> {
```

Comment:
No need for AtomicReference.
Consider replacing ifPresent with get. The map is already filtered on `node.allocation().isPresent()`